tests/test_training.py

```diff
@@ -183,7 +186,7 @@ def _test_checkpoints(self, training_loop_type: str, checkpoint_directory: str):
             num_epochs=int(self.num_epochs // 2),
             batch_size=self.batch_size,
             checkpoint_name=self.checkpoint_file,
-            checkpoint_directory=checkpoint_directory,
+            checkpoint_directory=self.temporary_directory,
```

Comment:
if you use TemporaryDirectory() as a context manager, the name is returned as the context. However if you save as a variable, you need `self.temporary_directory.name`